source/usage-examples/insertOne.txt

```diff
@@ -0,0 +1,54 @@
+=================
+Insert a Document
+=================
+
+.. default-domain:: mongodb
+
+You can insert a single document into a collection using the ``insertOne()``
+method on a ``MongoCollection`` object. To insert a document, construct a
+``Document`` object that contains the fields and values that you want to
+store. If you call the ``insertOne()`` method on a collection that does
+not exist yet, the method creates it for you.
+
+For more information on inserting data using MongoDB with the Java driver,
+see our guide (TODO: Fundamentals > CRUD Operations > Write Operations)
+on inserting documents.
+
+The method returns an instance of ``InsertOneResult`` which contains
+information such as the id (also known as the ``ObjectId``) of the
+inserted document or ``null`` if you specified a value for the ``_id``
+field in the inserted document.
```

Comment:
This sentence has a lot going on. From what I can tell:

- if the method successfully inserts a document, the method returns an object of type ``InsertOneResult``.
- That result contains information about the successful insertion operation
- One of those pieces of returned information is the ``_id`` of the inserted document IF you did NOT specify the _id when you inserted
- If you specified an ``_id`` for the inserted document, a successful insert will contain a value of ``null` for the ``_id`` field
- The ``_id`` returned by a successful operation is of type ``ObjectId``, or maybe is just also called "the ObjectId", or maybe is the ``InsertOneResult.ObjectId`` field of the returned object? (I think this is the only part of the thought that I would call truly confusing)

Overall I think all of the information here is good, but I think it would be easier to parse if we broke it up into some distinct thoughts -- particularly the types and rules surrounding the returned _id value. If possible, we should probably link to the API docs for each type as we mention them for further reading.

Reply:
Thanks for the feedback. I'll split up the sentence to make it more clear. As mentioned in the last sentence of the last paragraph before the example, the API doc links are located at the end of the page, based on Jeff Yemin's preference.